snow/engine/snowman/bootstrap/interval/state.go

```diff
@@ -0,0 +1,93 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package interval
+
+import (
+	"errors"
+
+	"github.com/ava-labs/avalanchego/database"
+)
+
+const (
+	rangePrefixByte byte = iota
+	blockPrefixByte
+
+	prefixLen = 1
+)
+
+var (
+	rangePrefix = []byte{rangePrefixByte}
+	blockPrefix = []byte{blockPrefixByte}
+
+	errInvalidKeyLength = errors.New("invalid key length")
+)
+
+func GetIntervals(db database.Iteratee) ([]*Interval, error) {
+	it := db.NewIteratorWithPrefix(rangePrefix)
+	defer it.Release()
+
+	var intervals []*Interval
+	for it.Next() {
+		dbKey := it.Key()
+		if len(dbKey) < prefixLen {
+			return nil, errInvalidKeyLength
+		}
+
+		rangeKey := dbKey[prefixLen:]
+		upperBound, err := database.ParseUInt64(rangeKey)
+		if err != nil {
+			return nil, err
+		}
+
+		value := it.Value()
+		lowerBound, err := database.ParseUInt64(value)
+		if err != nil {
+			return nil, err
+		}
+
+		intervals = append(intervals, &Interval{
+			LowerBound: lowerBound,
+			UpperBound: upperBound,
+		})
+	}
+	return intervals, it.Error()
+}
+
+func PutInterval(db database.KeyValueWriter, upperBound uint64, lowerBound uint64) error {
+	rangeKey := database.PackUInt64(upperBound)
+	return database.PutUInt64(
+		db,
+		append(rangePrefix, rangeKey...),
+		lowerBound,
+	)
+}
```

Comment:
We should document that the key is the `upperBound` because it is assumed that we typically fetch towards genesis. So, the `lowerBound` will be changed more frequently than the `upperBound`. This allows us to just overwrite the range rather than needing to delete the prior range and then writing the new range.